janusgraph-test/src/main/java/org/janusgraph/graphdb/JanusGraphTest.java

```diff
@@ -5680,4 +5767,39 @@ public void testAutoSchemaMakerForEdgePropertyDataType() {
             "Data type not identified correctly by auto schema maker");
     }
 
+    /* ==================================================================================
+                            IO
+     ==================================================================================*/
+
+    @Test
+    public void testWriteAndReadWithJanusGraphIoRegistryWithGryo(@TempDir Path tempDir) {
+        final Path file = tempDir.resolve("testgraph_" + this.getClass().getCanonicalName() + ".kryo");
+        testWritingAndReading(file.toFile());
+    }
+
+    @Test
+    public void testWriteAndReadWithJanusGraphIoRegistryWithGraphson(@TempDir Path tempDir) {
+        final Path file = tempDir.resolve("testgraph_" + this.getClass().getCanonicalName() + ".json");
+        testWritingAndReading(file.toFile());
+    }
+
+    private void testWritingAndReading(File f) {
+        graph.addVertex("name", f.getName());
```

Comment:
It would be better to use a traversal instead of graph, or?

Reply:
Yes, you're right of course. I just used the graph API because that's how the other tests in that class where doing it, but I now changed it to use the traversal API.